client/app/lib/contacts_listener.coffee

```diff
@@ -25,15 +29,54 @@ module.exports = class ContactsListener extends CozySocketListener
                 options = {}
             else
                 action  = 'add'
-                options = merge: true
-
-        @collection[action] @queues[queue], options
-        @queues[queue].length = 0
+                options =
+                    merge: true
+                    trigger: false
+
+
+        if @enabled
+            console.log """
+            Remote action occured (#{queue}, #{action}), processing start!
+            """
+
+            @collection.disableSort()
+            @end = new Date().getTime()
+
+            # Build batches of 10 models on which to perform operations.
+            lists = []
+            nbOccurences = @queues[queue].length
+            while @queues[queue].length > 0
+                lists.push @queues[queue].splice 0, 10
+
+            # Make a break between each operation performing to let the
+            # browser breath between each of them. This hack is dirty, we
+            # should find something better to make the massive addition less
+            # painful.
+            async.eachSeries lists, (list, done) =>
+                @collection[action] list, options
```

Comment:
Theorically, you can just pass set options.sort = true / false and remove the disableSort